### PR TITLE
feat: implement featured highlights list

### PR DESCRIPTION
**Pull request by OgDev-01**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
This PR implements a featured highlights list, adding a new feature to the project. The list will display important highlights or updates, making it easier for users to stay informed. This addition enhances the user experience and improves the overall functionality of the project.

_Generated using [OpenSauced](https://opensauced.ai/)._

## Related Tickets & Documents
This feature is tracked in #1289 

## Mobile & Desktop Screenshots/Recordings
image

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

File: pages/feed/index.tsx [hunk 1]
```diff
@@ -39,6 +41,12 @@ const Feeds: WithPageLayout<HighlightSSRProps> = (props: HighlightSSRProps) => {
   const { user } = useSupabaseAuth();
   const { data: repos } = useFetchHighlightRepos();
 
+  const { data: FeaturedHighlights } = useFetchFeaturedHighlights();
```
Comment (brandonroberts) [inline pages/feed/index.tsx:44]: nit: camel case variables

```suggestion
  const { data: featuredHighlights } = useFetchFeaturedHighlights();
```

File: pages/feed/index.tsx [hunk 2]
```diff
@@ -39,6 +41,12 @@ const Feeds: WithPageLayout<HighlightSSRProps> = (props: HighlightSSRProps) => {
...
+
+  const panelFormattedFeaturedHighlights = FeaturedHighlights?.map(({ id, title }) => ({
```
Comment (brandonroberts) [inline pages/feed/index.tsx:46]: The mapping isn't need here if you use `DbHighlight[]` as the type

File: components/molecules/FeaturedHighlightsPanel/featured-highlights-panel.tsx [hunk 3]
```diff
@@ -2,7 +2,7 @@ import Link from "next/link";
 import React, { useEffect, useState } from "react";
 
 interface FeaturedHighlightsPanelProps {
-  highlights: { id: number; title: string }[];
+  highlights: { id: string; title: string }[];
```
Comment (brandonroberts) [inline components/molecules/FeaturedHighlightsPanel/featured-highlights-panel.tsx:5]:
```suggestion
  highlights: DbHighlight[];
```

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: stories/molecules/featured-highlights-panel.stories.tsx [hunk 4]
```diff
@@ -1,19 +1,44 @@
...
 export default storyConfig;
 
-const sampleHighlights = [
-  { id: 1, title: "OAuth, never missing a chance to surprise! 🍕" },
-  { id: 106, title: "Replaced Supabase ui checkbox with Radix UI for better performance" },
-  { id: 106, title: "Even docs can experience breaking changes! 😅" },
+const sampleHighlights: DbHighlight[] = [
+  {
+    id: "1",
+    name: "OAuth, never missing a chance to surprise! 🍕",
```
Comment (NsdHSO) [inline stories/molecules/featured-highlights-panel.stories.tsx:13]: Just a little fun 👍 💯 
```suggestion
    name:'A Pizza Surprise: Unleashing the Power of OAuth! 🍕🔥',
```

**Comment by brandonroberts** [comment]
@OgDev-01 is there an associated issue/design for this?

**Comment by OgDev-01** [comment]
> @OgDev-01 is there an associated issue/design for this?

No we have none

**Comment by bdougie** [comment]
> > @OgDev-01 is there an associated issue/design for this?
> 
> No we have none

Link the tracking issue? Just don't prefix "close" or "fixes"

**Comment by bdougie** [comment]
I'd expect the "featured" highlight be orange without hover. 

Screen Shot 2023-07-10 at 4 45 52 PM

Rewatch GIF Recording - 2023-07-10 at 4 47 13 PM

Is this the correct intepretation @isabensusan?

**Review by bdougie — APPROVED** [review]

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**